lib/PSR/ContextBuilder.hs

```diff
@@ -49,46 +48,50 @@ import PlutusLedgerApi.V3.EvaluationContext qualified as V3
 -- Q: Is there a better way to represent this?
 
 -- Context0 is immediately derivable from the transactions.
-data Context0 where
+data Context0 era where
```

Comment:
I don't think this is a change we can make, making the era visible at the type level means we have to choose a specific era. The pipeline should be able to handle blocks from different eras, which we can branch on using the `ctxShelleyBasedEra` inside each processing function.

Since we want to be able to leash a node, presumably from genesis of the chain and capture all executions of a script ever, we need support for all eras.